system/jlib/jstats.cpp

```diff
@@ -396,6 +400,43 @@ static StringBuffer & formatIPV4(StringBuffer & out, unsigned __int64 value)
     return out.appendf("%d.%d.%d.%d", ip1, ip2, ip3, ip4);
 }
 
+static const char *getMoneyLocale()
+{
+    static bool haveLocale = false;
+    static StringAttr moneyLocale;
+
+    if (!haveLocale)
+    {
+        StringBuffer localestr;
+        if (isGlobalConfigReady())
+        {
+            queryGlobalConfig().getProp("cost/@moneyLocale", localestr);
+            haveLocale = true;
+            try // try setting locale, so exceptions can be caught here
+            {
+                std::locale(localestr.str());
+            }
+            catch(...)
+            {
+                OERRLOG("moneyLocale invalid: %s", localestr.str());
+                localestr.clear();
+            }
+        }
+        // note that an empty string causes default locale to used
+        moneyLocale.set(localestr.str());
+    }
+
+    return moneyLocale.get();
+}
+
+static StringBuffer & formatMoney(StringBuffer &out, unsigned __int64 value)
+{
+    std::stringstream ss;
+    ss.imbue(std::locale(getMoneyLocale()));
+    ss << std::showbase << std::put_money(value/1E4);
```

Comment:
Shouldn't this be cost_type2money(value)?

Reply:
No. put_money requires the money amount multiplied by 100.   Should I do  cost_type2money(value)*100 to make the code clearer?

Reply:
That sounds weird!
Looking at the examples on https://en.cppreference.com/w/cpp/io/manip/put_money I think it is implying that the money amount for the us is actually cents.  (The japanese yen example is not scaled.)
More details here:  https://www.stroustrup.com/3rd_loc.pdf
So you could either document that it is in cents, or multiply by 10^ locale...?..frac_digits